RIOT-OS/RIOT
# tests/periph/selftest_shield: fix integration

### Contribution description

The test should execute only with `make test-with-config` and not with `make test`, as boards without the shield cannot pass the test.

For some reason I accidentally added both variants, which makes no sense. This drops the `make test` variant.

Finally, the `README.md` is updated to refer to `make test-with-config`
instead of `make test`.

### Testing procedure

`make test/available` should now fail. As a result, the test should no longer execute automatically e.g. with `compile_and_test_for_board.py` and fail due to missing hardware.

`make test-with-config` should still execute the test.

### Issues/PRs references

None
